clients/ui/bff/internal/api/catalog_models_handler.go

```diff
@@ -0,0 +1,187 @@
+	"github.com/kubeflow/model-registry/ui/bff/internal/models"
+)
+
+type CatalogModelEnvelope Envelope[models.CatalogModel, None]
+type CatalogModelListEnvelope Envelope[models.CatalogModelList, None]
+type CatalogGroupedResponseEnvelope Envelope[CatalogSourceGroup, None]
+
+type CatalogSourceGroup struct {
+	Sources    []SourceGroup  `json:"sources"`
+	Pagination PaginationMeta `json:"pagination"`
+}
+
+type SourceGroup struct {
+	Source string                `json:"source"`
+	Models []models.CatalogModel `json:"models"`
+}
+
+type PaginationMeta struct {
+	ReturnedModels int32  `json:"returnedModels"`
+	PageSize       int32  `json:"pageSize"`
+	NextPageToken  string `json:"nextPageToken"`
+}
+
+func (app *App) GetAllCatalogModelsHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	ctxLogger := helper.GetContextLoggerFromReq(r)
+	ctxLogger.Info("This functionality is not implement yet. This is a STUB API to unblock frontend development")
+
+	source := r.URL.Query().Get("source")
+	if source == "" {
+		app.badRequestResponse(w, r, errors.New("source query parameter is required"))
+		return
+	}
+
+	query := r.URL.Query().Get("q")
+
+	// TODO: Implement actual catalog API call
+	// For now, return mock response to unblock frontend development
+	allMockModels := mocks.GetCatalogModelMocks()
+
+	var filteredModels []models.CatalogModel
+
+	if source == "ALL" || source == "all" {
+		filteredModels = allMockModels
+	} else {
+		for _, model := range allMockModels {
+			if model.SourceId != nil && *model.SourceId == source {
+				filteredModels = append(filteredModels, model)
+			}
+		}
+	}
+
+	// TODO: this is to unblock frontend development, will be removed in actual implementation
+	if query != "" {
+		var queryFilteredModels []models.CatalogModel
+		queryLower := strings.ToLower(query)
+
+		for _, model := range filteredModels {
+			matchFound := false
+
+			// Check name
+			if strings.Contains(strings.ToLower(model.Name), queryLower) {
+				matchFound = true
+			}
+
+			// Check description
+			if !matchFound && model.Description != nil && strings.Contains(strings.ToLower(*model.Description), queryLower) {
+				matchFound = true
+			}
+
+			// Check provider
+			if !matchFound && model.Provider != nil && strings.Contains(strings.ToLower(*model.Provider), queryLower) {
+				matchFound = true
+			}
+
+			// Check tasks
+			if !matchFound && model.Tasks != nil {
+				for _, task := range model.Tasks {
+					if strings.Contains(strings.ToLower(task), queryLower) {
+						matchFound = true
+						break
+					}
+				}
+			}
+
+			// Check license
+			if !matchFound && model.License != nil && strings.Contains(strings.ToLower(*model.License), queryLower) {
+				matchFound = true
+			}
+
+			if matchFound {
+				queryFilteredModels = append(queryFilteredModels, model)
+			}
+		}
+
+		filteredModels = queryFilteredModels
+	}
+
+	sourceGroups := groupModelsBySource(filteredModels)
```

Comment:
also, currently there are no API's to fetch/filter all the models from single api - i was thinking we could keep this aggregation of fetching models from all the sources, merge them and send it to the frontend with this format. 
does this approach make sense? curious to hear any suggestions or other ways to solve this.